packages/build-tools/src/steps/functions/installMaestro.ts

```diff
@@ -0,0 +1,190 @@
+import path from 'path';
+
+import fs from 'fs-extra';
+import {
+  BuildFunction,
+  BuildRuntimePlatform,
+  BuildStepGlobalContext,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+} from '@expo/steps';
+import spawn from '@expo/turtle-spawn';
+import { bunyan } from '@expo/logger';
+
+export function createInstallMaestroBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'install_maestro',
+    name: 'Install Maestro',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'maestro_version',
+        required: false,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+    ],
+    fn: async ({ logger, global }, { inputs, env }) => {
+      if (!(await isJavaInstalled())) {
+        if (
+          global.runtimePlatform === BuildRuntimePlatform.DARWIN &&
+          env.EAS_BUILD_RUNNER === 'eas-build'
+        ) {
+          logger.info('Installing Java');
+          await installJavaFromGcs({ logger });
+        } else {
+          // We expect Java to be pre-installed on Linux images,
+          // so this should only happen when running this step locally.
+          // We don't need to support installing Java on local computers.
+          throw new Error('Please install Java manually and rerun the job.');
+        }
+      }
+
+      // IDB is only a requirement on macOS.
+      if (global.runtimePlatform === BuildRuntimePlatform.DARWIN && !(await isIdbInstalled())) {
+        logger.info('Installing IDB');
+        await installIdbFromBrew({ global, logger });
+      }
+
+      // Skip installing if the input sets a specific Maestro version to install
+      // and it is already installed which happens when developing on a local computer.
+      const requestedMaestroVersion = inputs.maestro_version.value as string | undefined;
+      let currentMaestroVersion: string | undefined;
+      if (requestedMaestroVersion) {
+        try {
+          currentMaestroVersion = await getMaestroVersion();
+        } catch {}
+      }
+
+      if (!currentMaestroVersion || requestedMaestroVersion !== currentMaestroVersion) {
+        await installMaestro({
+          version: requestedMaestroVersion,
+          global,
+          logger,
+        });
```

Comment:
I'm hesitant whether we want to run it for people running custom builds locally (`eas build --local`) 🤔 

I mean I see a case where it can alter software installed on the user's machine, without the user wanting this software to be altered (even though they called the `install_maestro` function). I believe we should do the same thing as for Java here 🤔  

So ultimately this step should not install anything when running in local mode, but rather only tell people what they need to install manually and validate installed stuff in my opinion. 

If this would be something interactive it would be probably best to just prompt the user if they want to install XYZ, but we can't do stuff like this here. So I think it's better to not install software on the user's computer at all vs install it without the user truly wanting to have any new software installed on their device by us.

Maybe to make it more user-friendly we should give people a command they can run to install stuff in an error message.

Reply:
That's a reasonable concern. I'll make it so.

Reply:
https://github.com/expo/eas-build/blob/a2b2b25659139ce8b6f1c477d02182d2a8a55f78/packages/build-tools/src/steps/functions/installMaestro.ts#L31-L69